pages/api/v1/listing/on-chain-posts.ts

```diff
@@ -483,7 +490,12 @@ export async function getOnChainPosts(params: IGetOnChainPostsParams): Promise<I
 					variables: postsVariables
 				});
 			} catch (error) {
-				const data = await fetchSubsquare(network, Number(listingLimit), Number(page), Number(trackNo));
+				let data;
+				if (trackNo) {
```

Comment:
no need of if/else. The fetch subsquare function takes care of validations. Ideally another function should be there to where you simply pass the url, for now this if else is not required